### ISSUE TITLE
Orchestrator in Pending

### ISSUE THREAD

**Issue by andreujuanc** [Azure/azure-functions-durable-extension#1106]
### Description
Got some orchestrators that do not start and the status stays in Pending.
Service bus trigger creates X orchestartors with crafted Ids, they are unique (i checked), and not all of them are started, the amount of Pending is very inconsistent.

### App Details

- **Durable Functions extension version (e.g. v1.8.2)**:
- **Azure Functions runtime version (2.0)**:
- **Programming language used**: C#

### If deployed to Azure

- **Timeframe issue observed**: 
```
"createdTime": "2019-12-13T15:24:46Z",
"lastUpdatedTime": "2019-12-13T15:24:46Z"
```
- **Azure region**: West Europe
- **Orchestration instance ID(s)**: `000016654` `000016656` `000016658` `000016657`

I have the feeling that it might be caused by someone changing storage accounts, since this environment is the one used for integration testing.

But at the time of the test I did, there was no infrastructure change, and one of the request started, but the other 4 did not.

Thanks, 

Hope it's just something we are doing wrong.

**Comment by cgillum** [comment]
Hi @andreujuanc this looks weird. I took a look at `000016654` and it appears that two orchestrations were created at the exact same time, around 2019-12-13 15:24:47. In fact, I see in our Azure Storage logs that the exact same message was received twice from the same queue at almost the same time (one with a dequeue count of 1 and the other with a dequeue count of 2). I don't know exactly what to make of this, but it's probably causing the orchestration engine to get confused.

As an aside, I did notice that I'm seeing multiple storage accounts and task hubs associated with the instance ID, region (West EU), and time range you provided. Can you clarify which storage account name and/or task hub name where you were observing this problem? I want to make sure I'm looking at the right data.

Have you seen this problem since? If so, one thing you might want to try is adding a nuget dependency to the latest version of **Microsoft.Azure.DurableTask.AzureStorage** (it's only included by default in v2.1.0 of the extension). We've made quite a few reliability fixes there related to messaging, and your extension version (v1.8.2) will not have this latest version referenced.

**Comment by andreujuanc** [comment]
Hi @cgillum 

Sorry for the late reply. I cannot repro this, but we are in the middle of deploying it to Test environment, in which hopefully will be more stable environment, ie, noone is going to mess around with the azure resources.

I will try to upgrade to 2.1 of durable extensions before the release, cos I saw a bunch of closed issues there (great job, btw)

I will open a new issue if I encounter the same problem again.

Thanks
